### PR TITLE
Ewl 824 add initial choice to props

### PR DESCRIPTION
**Pull request by degouville**
# About the type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Test plan

- [ ] Empty data
- [ ] Invalid data
- [X] Correct data

### CODE REVIEW

File: src/components/Dropdown/index.js [hunk 1]
```diff
@@ -211,15 +215,18 @@ class Dropdown extends React.Component {
 
 Dropdown.defaultProps = {
   onItemSelected: null,
+  initialOptionID: null,
 };
 
+// TODO: Add initial choice
```
Comment (davidgarry) [inline src/components/Dropdown/index.js]: TODO ?

File: src/components/Dropdown/index.js [hunk 2]
```diff
@@ -156,8 +156,12 @@ class Dropdown extends React.Component {
   }
 
   setSelectedOption = () => {
-    const { options } = this.props;
-    this.setState({ selectedOption: options[0] });
+    const { options, initialOptionID } = this.props;
+    const matchInitialID = initialID => ({ id }) => id === initialID;
+    const initialOption = options.find(matchInitialID(initialOptionID));
+    const firstOption = options[0];
+    const selectedOption = initialOption || firstOption;
```
Comment (davidgarry) [inline src/components/Dropdown/index.js]: I think you can reduce the number of lines for that :
```
const initialOption = options.find(({ id }) => id === initialOptionID);
const selectedOption = initialOption || options[0];
```

File: src/components/Dropdown/index.js (hunk 1 above)
Comment (ra-phael) [inline src/components/Dropdown/index.js]: Can't this comment be removed now that initialOptionID was added?

File: src/components/Dropdown/index.stories.js [hunk 3]
```diff
@@ -2,8 +2,15 @@ import React from 'react';
...
 
 const dropdownStories = storiesOf('Components|Dropdown', module);
 
-dropdownStories.add('Default', () => <Dropdown options={object('options', dropdownOptions)} />);
+dropdownStories
+  .add('Default', () => <Dropdown options={object('options', dropdownOptions)} />)
+  .add('With Initial Option', () => (
+    <Dropdown
+      options={object('options', dropdownOptions)}
+      initialOptionID={object('InitialOptionID', initialOptionID)}
```
Comment (davidgarry) [inline src/components/Dropdown/index.stories.js]: This one is not an object knob

Comment (ra-phael) [inline src/components/Dropdown/index.stories.js]: Looks like it's a number, not an object 🤔

File: src/components/Dropdown/index.js (hunk 1 above)
Comment (degouville) [reply src/components/Dropdown/index.js]: yes

File: src/components/Dropdown/index.js (hunk 2 above)
Comment (degouville) [reply src/components/Dropdown/index.js]: ok

File: src/components/Dropdown/index.stories.js (hunk 3 above)
Comment (degouville) [reply src/components/Dropdown/index.stories.js]: Right.

FIXED ✅

File: src/components/Dropdown/index.js (hunk 1 above)
Comment (degouville) [reply src/components/Dropdown/index.js]: REMOVED ✅

File: src/components/Dropdown/index.stories.js (hunk 3 above)
Comment (degouville) [reply src/components/Dropdown/index.stories.js]: Right.

FIXED ✅

File: src/components/Dropdown/index.js [hunk 4]
```diff
@@ -156,8 +156,11 @@ class Dropdown extends React.Component {
   }
 
   setSelectedOption = () => {
-    const { options } = this.props;
-    this.setState({ selectedOption: options[0] });
+    const { options, initialOptionID } = this.props;
+    const matchInitialID = initialID => ({ id }) => id === initialID;
```
Comment (JulienKode) [inline src/components/Dropdown/index.js]: Should we test an option change ?

Comment (degouville) [reply src/components/Dropdown/index.js]: I've made a test case for initial setting [here](https://github.com/EurosportDigital/web-toolkit/blob/06db19856937b16791d354db28b6f545755282c3/src/components/Dropdown/index.test.js#L50)

By "option change" you mean to test *L172* like this ? :
```
  it('should display selected option after new selection', () => {
    const wrapper = ...

    wrapper.find(StyledLi).get(2).simulate('click');

    const isSelected = li => !!li.prop('isSelected');
    const selectedOption = wrapper.findWhere(isSelected);
    const selectedOptionText = selectedOption.text();
    expect(selectedOptionText).toBe('Men doubles ✓');
    wrapper.unmount();
  });
```

Comment (JulienKode) [reply src/components/Dropdown/index.js]: Oh perfect I've just see that `setSelectedOption` is call on mount

Comment (degouville) [reply src/components/Dropdown/index.js]: Maybe "setSelectedOption" should be "setInitalOption" as there is a confusion... What do you think ?

Comment (JulienKode) [reply src/components/Dropdown/index.js]: @degouville good idea :)

Comment (degouville) [reply src/components/Dropdown/index.js]: Done ✅

**Pull request merged**